microcluster/app.go

```diff
@@ -42,6 +42,10 @@ type Args struct {
 	StateDir    string
 	SocketGroup string
 
+	// Consumers of MicroCluster should provide a version to serve at /cluster/1.0
+	// This is required at daemon start.
+	Version string
```

Comment:
As discussed today, it may be worth enforcing that this has to be a `semver` by making it a defined type.

Reply:
I'm a little hesitant to do this until we know that we can expect a semver version from Micro{Ceph,OVN}. Requiring semver may also be irritating without much benefit in other Micro* projects, since we're requiring that this field be populated before the daemon can start.
Also, the [golang.org implementation](https://pkg.go.dev/golang.org/x/mod/semver) doesn't expose a public type; it just uses `string`.
Let me know what you'd prefer.